### PR TITLE
New org components

### PR DESCRIPTION
**Pull request by goruha**
## what
* Generate components docs from `cloudposse-terraform-components` org
* Update `PyGithub` pip package
* Added `docs/components/library/aws/spacelift/README.md` and `docs/components/library/aws/tgw/README.md`
* Fix links referenced to renamed components 
1. `/components/library/aws/aws-sso` -> `/components/library/aws/identity-center`
2. `/components/library/aws/eks/karpenter/` -> `/components/library/aws/eks/karpenter-controller/`
3. `components/library/aws/datadog-configuration` -> `components/library/aws/datadog-credentials`
4. `/components/library/aws/argocd-repo/` -> `/components/library/aws/argocd-github-repo/`
* Added redirect rule for renamed components
* Refactor `docs-collator` for components
* Added docs grouping (ex.: `glue`, `spacelift`, `tgw`)
* Added components preserve provider in name on docs site (ex.: `aws-config`)

## why
* Migrate components to new organization
* Version `2` support rate-limiting so script would not fail because hitting the rate
* This is the only place where it makes sense to put that docs because spacelift and tgw components now in new repos
* Avoid build failing because of broken links
* Preserve links indexed by search engine working
* Improve code quality
* As we have each component in a separate repo, we need a way how to group them at the docs site
* Some components need to preserve `aws-` in the interest of readability
*

### CODE REVIEW

File: docs/layers/software-delivery/eks-argocd/eks-argocd.mdx [hunk 1]
```diff
@@ -337,7 +337,7 @@ runs:
 ### Implementation
 
 - [`eks/argocd`](/components/library/aws/eks/argocd/): This component is responsible for provisioning [ArgoCD](https://argoproj.github.io/cd/).
-- [`argocd-repo`](/components/library/aws/argocd-repo/): This component is responsible for creating and managing an ArgoCD desired state repository.
+- [`argocd-repo`](/components/library/aws/eks/argocd-repo/): This component is responsible for creating and managing an ArgoCD desired state repository.
```
Comment (milldr) [inline docs/layers/software-delivery/eks-argocd/eks-argocd.mdx:340]: this isn't the correct path for argocd-repo

File: docs/components/library/aws/spacelift/README.md
Comment (milldr) [inline docs/components/library/aws/spacelift/README.md:1]: remove this file from git. I see that you already added it to .gitignore too

File: docs/components/library/aws/tgw/README.md
Comment (milldr) [inline docs/components/library/aws/tgw/README.md:1]: remove this file from git. I see that you already added it to .gitignore too

File: docs/components/library/aws/spacelift/README.md
Comment (goruha) [reply docs/components/library/aws/spacelift/README.md:1]: no. I added to gitignore negative rule allow to store this files in git
```
!/docs/components/library/aws/spacelift/README.md
```

File: docs/components/library/aws/tgw/README.md
Comment (goruha) [reply docs/components/library/aws/tgw/README.md:1]: no. I added to gitignore negative rule allow to store this files in git
```
!/docs/components/library/aws/tgw/README.md
```

**Review by milldr — APPROVED** [review]

**Pull request merged**